optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/bi/FilteringBiJoiner.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.core.impl.score.stream.bi;
+
+import java.util.function.BiPredicate;
+import java.util.function.Function;
+
+import org.optaplanner.core.impl.score.stream.common.JoinerType;
+
+public final class FilteringBiJoiner<A, B> extends AbstractBiJoiner<A, B> {
+
+    public FilteringBiJoiner(BiPredicate<A, B> filter) {
+        super(filter);
+    }
+
+    @Override
+    public Function<A, Object> getLeftMapping(int joinerId) {
```

Comment:
Just noticed (not a problem in this PR). This `joinerId` is not an ID, it's an index, for CombinedBiJoiners. That variable should be renamed to `joinerIndex` or just `index` (I 'd say the latter). Again, doesn't need to happen in this PR.

What the different between an ID and and index?
ID's have gaps, as elements get deleted. Examples of ID's: 0, 1, 2, 4, 7
Indexes don't have gaps, they would naturally change if elements get deleted (ussually they won't): Examples of index: 0, 1, 2, 3, 4, 5